packages/cli/src/commands/init/utils/repo.spec.ts

```diff
@@ -0,0 +1,73 @@
+import {configureGitSetup, logFailure, logSuccess, selectRepo} from "./repo";
+
+describe('Repo', () => {
+
+    it('clone should throw an error and call removeDir', async () => {
+        const removeDir = jest.fn(() => {});
+        const tasks = configureGitSetup(
+            "./",
+            { branch: "", repo: "" },
+            () => Promise.reject(),
+            () => Promise.reject(new Error()),
+            removeDir
+        );
+
+        try {
+            await tasks.run();
+        } catch (err) {
+            expect(removeDir.mock.calls.length).toBe(1);
+        }
+    });
+
+    it('checkout should throw an error and call removeDir', async () => {
+        const removeDir = jest.fn(() => {});
+        const tasks = configureGitSetup(
+            "./",
+            { branch: "", repo: "" },
+            () => Promise.resolve(),
+            () => Promise.reject(new Error()),
+            removeDir
+        );
+
+        try {
+            await tasks.run();
+        } catch (err) {
+            expect(removeDir.mock.calls.length).toBe(1);
+        }
+    });
+
+    it('should return correct repo for Angular', () => {
+       const angularRepo = selectRepo({'ds-framework': 'angular'});
+        expect(angularRepo).toStrictEqual({"branch": "main", "repo": "rangle/radius-angular"});
+    });
+
+    it('should return correct repo for React', () => {
+        const reactRepo = selectRepo({'ds-framework': 'react["css"]'});
+        expect(reactRepo).toBeDefined();
+    });
+
```

Comment:
Didn't understand what this was doing at first but its good that there's a spy for `log`. However if the message 'coming-soon...' changes, shouldnt we remove `console.log('coming soon ...'`?

Reply:
Yeah! It is just an extra check, checking `console.log` messages is a little bit an overkill and also that's not the main point of this test

Reply:
Yep. I agree 😄